src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -3381,77 +3390,87 @@ internal void ResetRunspaceState(ExecutionContext context)
             }
         }
 
-        internal static void SetSessionStateDrive(ExecutionContext context, bool setLocation)
+        private static bool TryInitSessionStateCurrentDrive(ExecutionContext context, string location)
```

Comment:
`location` seems to only be used in the error message here, seems wrong to provide this value if the code isn't actually using it outside of an error. I don't think an error should be reported in a `Try*` method as well, the caller should be in charge of dealing with the error condition with this type of pattern.